graph/src/ext/futures.rs

```diff
@@ -42,32 +44,79 @@ impl<F: Future, C: Fn() -> F::Error> Future for Cancelable<F, C> {
     }
 }
 
-#[derive(Debug)]
+/// A `CancelGuard` or `SharedCancelGuard`.
+pub trait CancelGuardTrait {
+    /// Adds `canceler` to the set being guarded.
+    fn add_canceler(&self, canceler: oneshot::Sender<()>);
+}
```

Comment:
Similar thought here on `canceler` -> `cancel_sender`.

This method is only used internally by the helpers, right? Maybe add a comment so that new API users don't get confused and try to use it directly.

It seems like this could also be implemented by types that are not scope guards. How about `CancelGuardTrait` -> `Canceler`?

Reply:
one more: `add_canceler` -> `add_cancel_sender`